daemon/config/config_linux.go

```diff
@@ -208,10 +208,6 @@ func (conf *Config) ValidatePlatformConfig() error {
 		return errors.Wrap(err, "invalid fixed-cidr-v6")
 	}
 
-	if _, ok := conf.Features["windows-dns-proxy"]; ok {
-		return errors.New("feature option 'windows-dns-proxy' is only available on Windows")
```

Comment:
Not sure how likely it is, but the only thing we could do is to (in the Windows variant of this) either return an error, or to log a Error / Warning to surface that the user is using an option that's no longer taken into account; given that the default is "true", that should probably only be relevant if they have it configured _and_ set to `false`; https://github.com/moby/moby/blob/43bbacb23926e678906d8094f1d66edebe82f4f1/daemon/config/config.go#L687-L693